### PR TITLE
feat: Update and gen proto for regen-ledger v4.0 upgrade

### PR DESCRIPTION
**Pull request by blushi**
Closes: #36 

- Pulling `regen` proto files from regen-ledger master (https://github.com/regen-network/regen-ledger/commit/cb4993b31ec0da80b9f4142162f5197d193b803f).
- Pulling `cosmos/app` and `cosmos/orm` proto files using https://github.com/cosmos/cosmos-sdk/releases/tag/api%2Fv0.1.0-alpha5 since it's the dependency that is being used in regen-ledger:
https://github.com/regen-network/regen-ledger/blob/cb4993b31ec0da80b9f4142162f5197d193b803f/x/data/go.mod#L8
https://github.com/regen-network/regen-ledger/blob/cb4993b31ec0da80b9f4142162f5197d193b803f/x/ecocredit/go.mod#L7
https://github.com/regen-network/regen-ledger/blob/cb4993b31ec0da80b9f4142162f5197d193b803f/proto/buf.gen.gogo.yaml#L5

Will tag v0.2.3 once that's merged and publish it to npm.

**Note for reviewers**: 
No need to review the full files diff but you can test this from the demo app in this repo: 
- from `packages/demo-app`, use [`yarn link`](https://classic.yarnpkg.com/en/docs/cli/link) to use the local version of @regen-network/api instead of 0.2.2
- In registry-server, you'll need to update `LEDGER_TENDERMINT_RPC` in `server/.env` to use http://194.37.81.19:26657/ instead (RPC endpoint for a temporary testnet with regen-ledger v4) and restart the server (CORS blocks the use of this endpoint directly without going through a proxy). 
- Then from `packages/demapp/src/App.tsx`, use `http://localhost:5000/ledger` to set the default `tmUrl`.
- Finally use some query/msg from `@regen-network/api/lib/generated/regen/ecocredit/v1` for instance within the demo app (Faucet on this testnet: `curl http://185.227.109.187:8000/faucet/<addr>`)

### CODE REVIEW

File: packages/api/scripts/protocgen.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ for dir in $PROTO_DIRS; do
     -I "proto" \
     -I "third_party/proto" \
     --plugin=${TS_PROTO_BIN} \
-    --ts_proto_opt="esModuleInterop=true,forceLong=long,useOptionals=true,outputTypeRegistry=true" \
+    --ts_proto_opt="esModuleInterop=true,forceLong=long,useOptionals=messages,outputTypeRegistry=true" \
```
Comment (flagrede) [inline packages/api/scripts/protocgen.sh:14]: What does this change do?

**Review by flagrede — APPROVED** [review]
LGTM !

I managed to get a response from `Projects` query from local `@regen-network/api/lib/generated/regen/ecocredit/v1/query` ,on your branch, following your instructions.

Projects query

File: packages/api/scripts/protocgen.sh (hunk 1 above)
Comment (blushi) [reply packages/api/scripts/protocgen.sh:14]: using `true` has just been deprecated in favor of `messages`, see https://github.com/stephenh/ts-proto docs but it does the same thing

**Review by haveanicedavid — APPROVED** [review]

**Comment by haveanicedavid** [comment]
tested locally - seems to work 👍

**Pull request merged**